### DISCUSSION TITLE
How to eager load with bind

### DISCUSSION THREAD

**Discussion by stephenafamo** [aarondl/sqlboiler#982] [Q&A] [answered]
If I want to bind to a structure like:

```go
type CustomUser struct {
	User       models.User `boil:",bind"`
	TotalPosts int         `boil:"total_posts"`
}
```

How do I user `qm.Load` to load relationships for the User model in a situation such as:

```go
var customUsers []*CustomUser
models.NewQuery(
    qm.Load(models.UserRels.Books), // this doesn't work
).Bind(ctx, db, *customUsers)
```

**Comment by kblomster** [comment]
I don't think there's a way to do that, unfortunately. I've also tried and failed. As far as I can tell from the code, eager loading relationship requires that the base `Bind()` target is a generated SQLBoiler model, so if you try to bind to a custom struct it simply won't work. It's not just the lack of the `R` struct (where the relationships go) that's the problem, there's also some deep reflection magic (I forget exactly where) that expects the target to be a model and fails if it isn't.

edit: right, it's [callLoadFunction in eager_load.go](https://github.com/volatiletech/sqlboiler/blob/master/queries/eager_load.go#L149) that makes things go wrong when you do this. I guess you could possibly work around this by making your `CustomUser` struct sufficiently like an actual SQLBoiler model, but that seems... fragile, to me.

**Comment by aarondl** [comment]
Yeah, currently there's no way to achieve this nicely. It looks  for the L struct and a function that does the eager loading on the generated model. If someone ever gets around to rewriting the eager loading stuff this could be a part of it maybe :D

**Reply by stephenafamo** [reply]
I've been thinking if when Go1.18 releases it would make sense to write sqlboiler/v5 using generics. At that point, perhaps also rewriting the eager loading logic would make sense.

**Reply by aarondl** [reply]
It's a very promising and exciting idea.

**Comment by stephenafamo** [comment, marked as answer]
Should work in `v4.12.0+`
